2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,203 @@
+This will be a local json file containing only the latest version of each module.
+The cache will be stored in the user path.
+There is no system cache that is shared.
+
+### Updating the cache
+
+A `Update-PSResourceCache` cmdlet will download the json file from the registered
+repositories.
+A hash check is performed to determine if the current cache needs to be updated.
+After the update is complete, it will compare installed versions with the cache
+and list out modules that are updatable.
+
+### Repository management
+
+`Register-PSResourceRepository` will allow for registering additional repositories.
+A `-Default` switch enables registering PSGallery should it be accidentally removed.
+The `-URL` will accept the HTTP address without the need to specify `/api/v2` as
+that will be assumed and discovered at runtime.
+A `-InstallationPolicy` switch accepts `Trusted` or `Untrusted` (default) indicating
+whether to prompt the user when installing resources from that repository.
+
+`Get-PSResourceRepository` will list out the registered repositories.
+
+`Set-PSResourceRepository` can be used to update a repository URL or installation policy.
+
+`Unregister-PSResourceRepository` can be used to un-register a repository.
+
+### Finding resources
+
+Current design of PowerShellGet is to have different cmdlets for each of the
+different resources types that it supports:
+
+`Find-Command`
+`Find-DscResource`
+`Find-Module`
+`Find-RoleCapability`
+`Find-Script`
+
+Instead, these will be abstracted by a `Find-PSResource` cmdlet.
+A `-Type` parameter will accept an array of types to allow filtering.
+
+With support of the generic `PSResource`, this means we can also find and
+install arbitrary nupkgs that may only contain an assembly the user wants to
+use in their script.
+
+If the local cache does not exist, the cmdlet will call `Update-PSResourceCache`
+first and then attempt a search.
+However, this cmdlet does not update the cache if one exists.
+
+### Installing resources
+
+`Install-PSResource` will only work for modules.
+A `-AllowPrerelease` switch allows installing prerelease versions.
+Other types will use `Save-PSResource` (see below).
+A `-Repository` parameter accepts a specific repository name.
+If `-Repository` is not specified and the same resource is found in multiple
+repositories, then the resource is automatically installed quietly from the
+trusted repository with the highest version matching the `-Version` parameter
+(if specified, otherwise newest non-prerelease version unless `-AllowPrerelease`
+is used).
+If there are no trusted repositories matching the query, then the newest version
+fulfilling the query will be prompted to be installed.
+`-AllowUntrusted` can be used to suppress being prompted for untrusted sources.
+`-AllowDifferentPublisher` can be used to suppress being prompted if the publisher
+of the module is different from the currently installed version.
+`-AllowClobber` can be used to re-install a module.
```

Comment:
how about `-reinstall`?

Reply:
-reinstall makes sense if it's installing over an existing resource.  If it's a "force" type of behavior -reinstall might be a bit misleading

Reply:
`-AllowClobber` is already used in a few cmdlets.  `apt` uses `--reinstall` with `install` command.  I'm trying to move away from the non-descriptive `-Force` and so `-AllowReinstall` might be ok.

Reply:
So the major complaint around PSGet 2.0 was that you have to specify multiple switches to actually force. It wasn't the name of the switches, it was the quantity. To that end, folks like Jaykul pushed for a `-ForceAll` switch that *just installs the damn module*. I'm unopposed to throwing warnings when using that switch, but I feel pretty strongly we need one of those. 

More specifically, `-AllowClobber` as a name is more about multiple modules squatting on the same cmdlet names. It doesn't make sense to me as a reinstall parameter. While the implementation may look roughly the same, I think it's important we include an `-AllowReinstall` or `-Reinstall` in addition to `-AllowClobber`.